src/couch_index/src/couch_index_server.erl

```diff
@@ -203,6 +210,9 @@ handle_info({'EXIT', Pid, Reason}, Server) ->
                     ets:match_object(Server#st.by_db, {DbName, {'$1', Sig}})
             ],
             rem_from_ets(DbName, Sig, DDocIds, Pid, Server);
+        [{_, opening, {DbName, Sig}}] when Reason /= normal ->
+            Msg = {async_error, {DbName, nil, Sig}, Reason},
+            {reply, ok, _} = handle_call(Msg, {Pid, nil}, Server);
```

Comment:
This mirrors the pattern in dreyfus manager, but wonder if it makes sense to clean it up and just have a separate handle_async_error(DbName, Sig, OpenerPid, Error) function to call from `handle_call/3` and when opener crashes.